rollingpin/harold.py

```diff
@@ -63,7 +66,13 @@ def make_request(self, path, data):
             "Content-Type": ["application/x-www-form-urlencoded"],
             "X-Hub-Signature": ["sha1=" + body_producer.hash(self.secret)],
         })
-        return self.agent.request("POST", url, headers, body_producer)
+        req = self.agent.request("POST", url, headers, body_producer)
+
+        def log_timeout(d, timeout):
+            self.log.warning("harold: request timed out after %d seconds (/%s)", timeout, path)
```

Comment:
So i discovered this `onTimeoutCancel` callback API, and I decided to use it to log the timeout exception rather than following the [swallow_exceptions](https://github.com/reddit/rollingpin/blob/master/rollingpin/utils.py#L23-L23) pattern, mainly because at that point, the exception we get is a `CancelledError` which technically could be raised by something thats not an exception.

Adding another error callback here to filter the CancelledError into a TimeoutError was getting pretty hairy, because the failure is a ResponseFailed wrapper that needed to be unwrapped etc., which was way more complicated than just adding a log line here.

Reply:
:+1: